build_tools/azure/install.sh

```diff
@@ -55,7 +55,7 @@ pre_python_environment_install() {
 check_packages_dev_version() {
     for package in $@; do
         package_version=$(python -c "import $package; print($package.__version__)")
-        if ! [[ $package_version =~ "dev" ]]; then
+        if [[ $package_version =~ "\.[0-9]+$" ]]; then
```

Comment:
I don't understand the logic here. Shouldn't `\.` be under the scope of the `+` operator?

Also why a `$` without an `^`?

I would have expected:

```suggestion
        if [[ $package_version =~ "^[\.0-9]+$" ]]; then
```

I confirmed it with https://regex101.com/r/O54PWi/1

Reply:
This was only checking the last part after the dot, dot included, i.e. `.0a1` in `3.1.0a1`. I agree that your regex is more explicit. I'll push a fix doing this with the simplification that you can use `.` instead of `\.` inside `[]`.

Side-comment: the reason that this check needs to change is that Cython development version does not have dev in its name and is actually 3.1.0a1.